adapters/oidc/js/src/main/resources/keycloak.js

```diff
@@ -432,21 +436,19 @@
             }
 
             var baseUrl;
-            if (options && options.action == 'register') {
+            if (options && options.action === 'register') {
                 baseUrl = kc.endpoints.register();
             } else {
                 baseUrl = kc.endpoints.authorize();
             }
 
-            var scope;
-            if (options && options.scope) {
-                if (options.scope.indexOf("openid") != -1) {
-                    scope = options.scope;
-                } else {
-                    scope = "openid " + options.scope;
-                }
-            } else {
+            var scope = options && options.scope || kc.scope || config.scope;
```

Comment:
we check the scope parameter for:
- `options` are the login options
- `kc.scope` potentially derived from `keycloak.json`
- `config.scope` the explicit keycloak configuration.

Reply:
It would be much better IMHO to create a dedicated config object from `keycloak.json` instead of "parking" the configuration in the `kc` keycloak instance, but this would require a larger refactoring...

Reply:
Yeah, this is certainly something that is hard to follow as it is now. Should be re-factored in the future.